bindings/cs/MongoDB.Libmongocrypt/EncryptionAlgorithm.cs

```diff
@@ -1,24 +0,0 @@
-﻿/*
- * Copyright 2019–present MongoDB, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-namespace MongoDB.Libmongocrypt
```

Comment:
this change can be considered as BC, since in theory this is a public class, but it's public only because it's a different library, from user's perspective it's an internal class, so I think there is no harm to remove this and recreate on the driver side (with different `ns`)

Reply:
This is a public type and moving it to another assembly and changing namespaces is a breaking change. Note that the following currently compiles and runs:

```
dotnet new console
dotnet add package MongoDB.Driver
```

```
using System;
using MongoDB.Libmongocrypt;
var algorithm = EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic;
Console.WriteLine(algorithm);
```

Even though this public type is in a referenced assembly, users could be depending on it.

Same thing with changing the parameter type of `CryptClient.StartExplicitEncryptionContextWithKeyId` (and related) from `EncryptionAlgorithm` to `string`.

I know that it is more painful to have to overload methods and deprecate the old methods/classes, but I don't see that we have much choice.